Handle empty return for protocols with no datasources that have no configuration saved

```bash
2016-01-05 12:18:31,157 [ERROR] django.request: Internal Server Error: /portal/api/protocols/15/data_sources/
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 112, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python2.7/site-packages/django/views/decorators/csrf.py", line 57, in wrapped_view
    return view_func(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/rest_framework/viewsets.py", line 85, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/rest_framework/views.py", line 452, in dispatch
    response = self.handle_exception(exc)
  File "/usr/local/lib/python2.7/site-packages/rest_framework/views.py", line 449, in dispatch
    response = handler(request, *args, **kwargs)
  File "/opt/app/brp/api/views.py", line 116, in data_sources
    dc = json.loads(pds.driver_configuration)
  File "/usr/local/lib/python2.7/json/__init__.py", line 338, in loads
    return _default_decoder.decode(s)
  File "/usr/local/lib/python2.7/json/decoder.py", line 366, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/local/lib/python2.7/json/decoder.py", line 384, in raw_decode
    raise ValueError("No JSON object could be decoded")
ValueError: No JSON object could be decoded
```

--- a/brp/api/fixtures/test_data.json
+++ b/brp/api/fixtures/test_data.json
@@ -40,6 +40,15 @@
         "created": "2015-09-29T12:01:41.671"
     }
 }, {
+    "pk": 3,
+    "model": "portal.immutablekey",
+    "fields": {
+        "modified": "2015-09-29T12:01:41.672",
+        "key": "BOGUSIMMUTABLEKEY",
+        "created": "2015-09-29T12:01:41.671"
+    }
+},
+{
     "pk": 1,
     "model": "portal.organization",
     "fields": {
@@ -76,6 +85,20 @@
     }
 }, {
     "pk": 1,
+    "model": "portal.protocoldatasource",
+    "fields": {
+        "data_source": 1,
+        "protocol": 1,
+        "created": "2015-09-29T12:04:57.702",
+        "max_records_per_subject": -1,
+        "driver": 0,
+        "modified": "2015-09-29T12:04:57.702",
+        "display_label": "Health Data",
+        "path": "Demo",
+        "driver_configuration": ""
+    }
+}, {
+    "pk": 1,
     "model": "portal.protocol",
     "fields": {
         "organizations": [1],
@@ -85,6 +108,17 @@
         "created": "2015-09-29T12:01:41.672"
     }
 }, {
+    "pk": 2,
+    "model": "portal.protocol",
+    "fields": {
+        "organizations": [1],
+        "immutable_key": 3,
+        "modified": "2015-09-29T12:01:41.672",
+        "name": "Protocol Without Datasources",
+        "created": "2015-09-29T12:01:41.672"
+    }
+},
+{
     "pk": 1,
     "model": "portal.protocoluser",
     "fields": {

--- a/brp/api/tests.py
+++ b/brp/api/tests.py
@@ -19,3 +19,44 @@ class TestProtocolDatasourceAPI(APITestCase):
         self.client.credentials(HTTP_AUTHORIZATION='Token ' + token.key)
         response = self.client.get(url, format='json')
         self.assertTrue(True, response.has_header('content-length'))
+
+
+class TestProtocolAPI(APITestCase):
+
+    fixtures = ['brp/api/fixtures/test_data.json']
+
+    def test_protocol_with_no_datasources(self):
+        token = Token.objects.get(user__username='admin')
+        url = reverse('protocol-datasources-list', args=[2])
+        self.client.credentials(HTTP_AUTHORIZATION='Token ' + token.key)
+        response = self.client.get(url, format='json')
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual([], response.data)
+
+    def test_protocol_with_empty_datasource_configuration(self):
+        token = Token.objects.get(user__username='admin')
+        url = reverse('protocol-datasources-list', args=[1])
+        self.client.credentials(HTTP_AUTHORIZATION='Token ' + token.key)
+        response = self.client.get(url, format='json')
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            [{
+                'data_source': {
+                    'description': u"CHOP's REDCap Instance",
+                    'url': u'https://redcap.chop.edu/api/',
+                    'ehb_service_es_id': 1,
+                    'desc_help': 'Please briefly describe this data source.',
+                    'id': 1,
+                    'name': u'REDCap'
+                },
+                'protocol': 'http://testserver/api/protocols/1/',
+                'max_records_per_subject': -1,
+                'driver': 0,
+                'driver_configuration': '',
+                'display_label': u'Health Data',
+                'authorized': True,
+                'path': u'Demo',
+                'id': 1
+            }],
+            response.data
+        )

--- a/brp/api/views.py
+++ b/brp/api/views.py
@@ -113,7 +113,10 @@ class ProtocolViewSet(viewsets.ModelViewSet):
         for pds in p.getProtocolDataSources():
             t = ProtocolDataSourceSerializer(pds, context={'request': request}).data
             # Parse ProtocolDataSource configuration
-            dc = json.loads(pds.driver_configuration)
+            if pds.driver_configuration != '':
+                dc = json.loads(pds.driver_configuration)
+            else:
+                dc = ''
             t["driver_configuration"] = dc
             # Determine Authorization
             try:

